### PR TITLE
Add sealed trait analysis deep-dive for cargo-semver-checks v0.35

### PR DESCRIPTION
**Pull request by obi1kenobi**
This post describes the inner workings of the headline feature for `cargo-semver-checks v0.35`.

Improving `cargo-semver-checks` is [an official Rust project goal](https://rust-lang.github.io/rust-project-goals/2024h2/cargo-semver-checks.html), so perhaps this post is a better fit for the "Project & Tooling Updates" section? I'll leave the decision up to the TWiR mods, I don't have a strong preference.

Thank you for running this awesome community resource!

### CODE REVIEW

File: draft/2024-09-04-this-week-in-rust.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ and just ask the editors to select the category.
 ### Project/Tooling Updates
 
 ### Observations/Thoughts
+- [Is this trait sealed, or not sealed — that is the question](https://predr.ag/blog/is-this-trait-sealed-or-not-sealed/)
```
Comment (mariannegoldin) [inline draft/2024-09-04-this-week-in-rust.md:41]:
```suggestion
* [Is this trait sealed, or not sealed — that is the question](https://predr.ag/blog/is-this-trait-sealed-or-not-sealed/)
```

**Review by mariannegoldin — APPROVED** [review]
TY! Approved with a formatting fix.

**Comment by mariannegoldin** [comment]
Whoops, I missed that we already have this in the existing TWIR via the editorial team. Closing this PR in lieu of that.

**Comment by obi1kenobi** [comment]
Thank you!

**Pull request closed without merging**